core/VectorUopGenerator.cpp

```diff
@@ -117,8 +150,10 @@ namespace olympia
         // TODO: For now, generate uops for all elements even if there is a tail
         num_uops_to_generate_ = std::ceil(vector_config->getVLMAX() / num_elems_per_uop);
 
-        if ((uop_gen_type == InstArchInfo::UopGenType::WIDENING)
-            || (uop_gen_type == InstArchInfo::UopGenType::MAC_WIDE))
+        if (uop_gen_type == InstArchInfo::UopGenType::WIDENING
+            || uop_gen_type == InstArchInfo::UopGenType::WIDENING_MIXED
+            || uop_gen_type == InstArchInfo::UopGenType::MAC_WIDE
+            || uop_gen_type == InstArchInfo::UopGenType::NARROWING)
         {
             // TODO: Add parameter to support dual dests
```

Comment:
It would be a good idea to add a `sparta_assert` to make sure that LMUL <= 4 for these categories.